app/models/framework/management_charge_calculator/flat_rate.rb

```diff
@@ -21,12 +21,23 @@ def calculate_for(entry)
       private
 
       def entry_value(entry)
-        if @column.present?
-          entry.data[@column]
+        if overridding_calculation_column?
+          cast_to_decimal(entry.data[@column])
         else
           entry.total_value
         end
       end
+
+      def overridding_calculation_column?
+        @column.present?
+      end
+
+      def cast_to_decimal(column_value)
+        case column_value
```

Comment:
What happens if the `column_value` is `nil`?

Reply:
Good question! The code would explode with a method-on-nil error. I did have a version of the code that checked for nil and raised a specific error, but removed it as the value should never be `nil` if the system is functioning as expected; if it was, it would get caught at the validation stage, before we get to calculating the charge. If the system _is_ trying to calculate the charge on an entry that is missing a required field, then there is something seriously wrong and we probably want it to fall over noisily.